src/samples/test-bots/TeamsSSOAgent/AuthAgent.cs

```diff
@@ -0,0 +1,180 @@
+
+        // If the sign in is successful, the user will be signed in and the token will be available from the UserAuthorization.GetTurnToken(DefaultHandlerName) call. 
+        // If the sign in was not successful, this won't be reached.  Instead, OnUserSignInFailure route would have been called. 
+        
+        // We have the access token, now try to get your user name from graph. 
+        string displayName = await GetDisplayName();
+        if (displayName.Equals(_defaultDisplayName))
+        {
+            // Handle error response from Graph API
+            await turnContext.SendActivityAsync($"Failed to get user information from Graph API \nDid you update the scope correctly in Azure bot Service?. If so type in -signout to force signout the current user", cancellationToken: cancellationToken);
+            return;
+        }
+
+        // Now Echo back what was said with your display name. 
+        await turnContext.SendActivityAsync($"**{displayName} said:** {turnContext.Activity.Text}", cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// This method is called when the sign-in process fails with an error indicating why . 
+    /// </summary>
+    /// <param name="turnContext"></param>
+    /// <param name="turnState"></param>
+    /// <param name="handlerName"></param>
+    /// <param name="response"></param>
+    /// <param name="initiatingActivity"></param>
+    /// <param name="cancellationToken"></param>
+    /// <returns></returns>
+    private async Task OnUserSignInFailure(ITurnContext turnContext, ITurnState turnState, string handlerName, SignInResponse response, IActivity initiatingActivity, CancellationToken cancellationToken)
+    {
+        // Raise a notification to the user that the sign-in process failed.
+        await turnContext.SendActivityAsync($"Sign In: Failed to login to '{handlerName}': {response.Cause}/{response.Error.Message}", cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// Gets the display name of the user from the Graph API using the access token.
+    /// </summary>
+    private async Task<string> GetDisplayName()
+    {
+        string displayName = _defaultDisplayName;
+        var graphInfo = await GetGraphInfo(UserAuthorization.DefaultHandlerName);
+        if (graphInfo != null)
+        {
+            displayName = graphInfo!["displayName"].GetValue<string>();
+        }
+        return displayName;
+    }
+
+    private async Task<JsonNode> GetGraphInfo(string handleName)
+    {
+        string accessToken = UserAuthorization.GetTurnToken(handleName);
+        string graphApiUrl = $"https://graph.microsoft.com/v1.0/me";
+        try
+        {
+            using HttpClient client = new();
```

Comment:
avoid new HttpClient, instead use the http factory from DI

Reply:
This is POC, Rido... not the final deal.  It is for comment on what it's doing.